content/en/docs/upgrade/devops-data-migration-upgrade.md

```diff
@@ -0,0 +1,104 @@
+
+### DevOps in KubeSphere v3.0
+
+![devops-3-0-architecture](/images/docs/upgrade/devops-data-migration-upgrade/devops-3-0-architecture.png)
+
+In KubeSphere v3.0, the DevOps data are stored in CustomResourceDefinition (CRD). The data in the CRD are synchronized to Jenkins in an one-way direction, and any previous data with the same name will be overwritten. There are mainly two types of operation. One type is the operation of creation, and the other is the operation of triggering.
+
+The operation of creation mainly involves three CRD types, including DevOps projects, pipelines and credentials. Users make calls to the interface of ks-apiserver through frontend to create resources and store them in the etcd, and then the ks-controller-manager keeps synchronizing those objects to Jenkins. The operation of triggering mainly involves instant actions, including running a pipeline and reviewing a pipeline. Users make calls to the ks-apiserver through frontend, and then, after data conversion, directly make calls to the Jenkins API.
+
+## The Data Migration Logic
+
+When upgrading KubeSphere from v2.1 to v3.0, the process of data migration follows the steps shown below:
+
+1. Retrieve information about DevOps directories from MySQL
+2. Retrieve information about pipelines and credentials from Jenkins
+3. Serialize the above information into a YAML file
+4. Back up the YAML resource object to the MinIO object storage
+5. Use the `kubectl apply` command to apply those pipeline resources
+6. Authorizations migration
+
+Data comparison before and after the migration:
+
+| Resource Affected |  v2.1   | v3.0 |
+| :---------------: | :-----: | :--: |
+|  DevOps projects  |  MySQL  | CRD  |
+|     Pipelines     | Jenkins | CRD  |
+|    Credentials    | Jenkins | CRD  |
+|  Authorizations   |  MySQL  | CRD  |
+
+{{< notice warning>}}
+
+You must back up the data in advance because the data in Jenkins will be overwritten by the ks-controller-manager once the CRD is created during data migration. You can use Velero to make a PVC-level backup, or back up the `/var/jenkins_home` directory of the Jenkins Pod directly.
+
+{{</ notice >}}
+
+## Data Migration Plans and Risks Evaluation
+
+For the pipelines created on the KubeSphere DevOps page, migration can be carried out as normal. However, the pipeline configurations directly made on the Jenkins dashboard might get lost during the migration.
+
+### Migration through a Job
+
+Follow the upgrade process as normal. This plan applys to the scenario where the DevOps operations are only made on the KubeSphere DevOps page. If there is no DevOps resources available after the upgrade, you can rerun the upgrade process.
+
+### No migration for the previous data
+
+KubeSphere DevOps is compatible with most Jenkins configurations, but not all of them. As shown in the below image, certain configurations on the Jenkins dashboard are not supported by KubeSphere DevOps.
```

Comment:
->
KubeSphere DevOps is compatible with most Jenkins configurations, while some on the Jenkins dashboard are not supported by KubeSphere DevOps as shown below: